src/adapters/Cornerstone3D/EllipticalROI.js

```diff
@@ -24,60 +30,74 @@ class EllipticalROI {
-        const minorAxisStart = { x: GraphicData[4], y: GraphicData[5] };
-        const minorAxisEnd = { x: GraphicData[6], y: GraphicData[7] };
-
-        const majorAxisIsHorizontal = majorAxisStart.y === majorAxisEnd.y;
-        const majorAxisIsVertical = majorAxisStart.x === majorAxisEnd.x;
-
-        let ellipsePointsImage;
-
-        if (majorAxisIsVertical) {
-            ellipsePointsImage = [
-                majorAxisStart.x,
-                majorAxisStart.y,
-                majorAxisEnd.x,
-                majorAxisEnd.y,
-                minorAxisStart.x,
-                minorAxisStart.y,
-                minorAxisEnd.x,
-                minorAxisEnd.y
-            ];
-        } else if (majorAxisIsHorizontal) {
-            ellipsePointsImage = [
-                minorAxisStart.x,
-                minorAxisStart.y,
-                minorAxisEnd.x,
-                minorAxisEnd.y,
-                majorAxisStart.x,
-                majorAxisStart.y,
-                majorAxisEnd.x,
-                majorAxisEnd.y
-            ];
-        } else {
-            throw new Error("ROTATED ELLIPSE NOT YET SUPPORTED");
-        }
-
-        const points = [];
-        for (let i = 0; i < ellipsePointsImage.length; i += 2) {
+        // in the image plane and then choose the correct points to use for the ellipse.
+        const pointsWorld = [];
+        for (let i = 0; i < GraphicData.length; i += 2) {
             const worldPos = imageToWorldCoords(imageId, [
-                ellipsePointsImage[i],
-                ellipsePointsImage[i + 1]
+                GraphicData[i],
+                GraphicData[i + 1]
             ]);
 
-            points.push(worldPos);
+            pointsWorld.push(worldPos);
+        }
+
+        const majorAxisStart = vec3.fromValues(...pointsWorld[0]);
+        const majorAxisEnd = vec3.fromValues(...pointsWorld[1]);
+        const minorAxisStart = vec3.fromValues(...pointsWorld[2]);
+        const minorAxisEnd = vec3.fromValues(...pointsWorld[3]);
+
+        const majorAxisVec = vec3.create();
+        vec3.sub(majorAxisVec, majorAxisEnd, majorAxisStart);
+
+        const minorAxisVec = vec3.create();
+        vec3.sub(minorAxisVec, minorAxisEnd, minorAxisStart);
+
+        const imagePlaneModule = metadata.get("imagePlaneModule", imageId);
+
+        if (!imagePlaneModule) {
+            throw new Error("imageId does not have imagePlaneModule metadata");
+        }
+
+        const { columnCosines } = imagePlaneModule;
+
+        // find which axis is parallel to the columnCosines
+        const columnCosinesVec = vec3.fromValues(...columnCosines);
+
+        const projectedMajorAxisOnColVec = vec3.dot(
+            columnCosinesVec,
+            majorAxisVec
+        );
+
+        const projectedMinorAxisOnColVec = vec3.dot(
+            columnCosinesVec,
+            minorAxisVec
+        );
+
+        let ellipsePoints = [];
+        if (projectedMajorAxisOnColVec < EPSILON) {
```

Comment:
Don't we have to take the absolute value of the dot product here? What if the major axis is along the direction cosine, but it points in the opposite direction?

Reply:
Also this if statement is saying "If the major axis _doesn't_ project on the column vector, insert it along the row (left&right coordinates of ellipse)", followed by another if that says "If the minor axis _doesn't_ project on the column vector, insert it along the row (left&right coordinates of ellipse)".

These negative statements are a bit confusing, its generally an antipattern to do "if not this, else this", when it can be better read as "if this, else this" by swapping the statements around.

It would be better to check if "if the major axis _does_ project on the column vector, insert it along the column vector (top&bottom coordinates of ellipse)", which can be done by the comparison:

```
// Normalize the majorAxisVec earlier
vec3.normalize(majorAxisVec, majorAxisVec)

// This can now only ever be a number from -1 to 1
const projectedMajorAxisOnColVec = vec3.dot(
      columnCosinesVec,
      majorAxisVec
);

// Take absolute.
const absoluteOfMajorDotProduct = Math.abs(projectedMajorAxisOnColVec)


if (Math.abs(absoluteOfMajorDotProduct - 1) < EPSILON) {
        // ... major axis vec is along column cosines vec...
}
```